### PR TITLE
[cppad/cpuid] Update to latest version

### LINKED ISSUE #19872: [CppAD] update to 20210000.7
Library name: CppAD

New version number: 20210000.7

Other information that may be useful (release notes, etc...)
https://github.com/coin-or/CppAD/releases

### LINKED ISSUE #19873: [cpuid] update to 0.5.1
Library name: cpuid

New version number: 0.5.1

Other information that may be useful (release notes, etc...)
https://github.com/anrieff/libcpuid/releases/latest

### PR DESCRIPTION
**Pull request by Cheney-W**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes https://github.com/microsoft/vcpkg/issues/19872#event-5275764601
  Fixes https://github.com/microsoft/vcpkg/issues/19873#event-5275764829

No feature need to be tested.

### CODE REVIEW

File: ports/cpuid/portfile.cmake [hunk 1]
```diff
@@ -4,25 +4,23 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
...
     HEAD_REF master
-    PATCHES fix-install-headers.patch
 )
 
-vcpkg_configure_cmake(
-        SOURCE_PATH ${SOURCE_PATH}
-        PREFER_NINJA
-        OPTIONS
+vcpkg_cmake_configure(
+    SOURCE_PATH ${SOURCE_PATH}
```
Comment (PhoebeHui) [inline ports/cpuid/portfile.cmake:13]:
```suggestion
    SOURCE_PATH "${SOURCE_PATH}"
```

File: ports/cppad/portfile.cmake [hunk 2]
```diff
@@ -5,17 +5,15 @@ vcpkg_fail_port_install(ON_TARGET "uwp" and "arm")
...
+    SHA512 112a4663a3e13f2d852c4ce4e57f6bee2dc7584915fcbab75972568258faab0d4a5761c4eaa4c664543cb8674e8e70c0623054c07dff933f9513a47f1c7d6261
     HEAD_REF master
     PATCHES
         windows-fix.patch
-        pkgconfig-fix.patch
 )
 
-vcpkg_configure_cmake(
+vcpkg_cmake_configure(
     SOURCE_PATH ${SOURCE_PATH}
```
Comment (PhoebeHui) [inline ports/cppad/portfile.cmake:16]:
```suggestion
    SOURCE_PATH "${SOURCE_PATH}"
```

File: ports/cpuid/portfile.cmake (hunk 1 above)
Comment (Cheney-W) [reply ports/cpuid/portfile.cmake]: Done.

File: ports/cppad/portfile.cmake (hunk 2 above)
Comment (Cheney-W) [reply ports/cppad/portfile.cmake]: Done.

**Review by PhoebeHui — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the updates!

**Pull request merged**